biorbd_optim/plot.py

```diff
@@ -161,16 +161,23 @@ def __init__(self, ocp, sol):
         self.sol = sol
 
     def show_biorbd_viz(self, show_meshes=True):
-        if self.ocp.nlp[0]["problem_type"] == ProblemType.torque_driven:
-            x, _, _ = ProblemType.get_data_from_V(self.ocp, self.sol["x"])
-
-        if self.ocp.nlp[0]["problem_type"] == ProblemType.muscles_and_torque_driven:
-            x, _, _, _ = ProblemType.get_data_from_V(self.ocp, self.sol["x"])
+        x = ProblemType.get_q_from_V(self.ocp, self.sol["x"])
 
         if self.ocp.nb_phases == 1:
             x = [x]
-        # x = self.ocp.nlp[0]["q_mapping"].expand(x)
 
+        else:
+            same_dof = True
+            for k in range(self.ocp.nlp[0]["model"].nbDof()):
+                if self.ocp.nlp[0]["model"].nameDof()[k].to_string() != self.ocp.nlp[1]["model"].nameDof()[k].to_string():
+                    same_dof = False
+                    break
+            if same_dof:
+                x_concat = x[0]
```

Comment:
Concat doesn't work. The last node of x[i] is the first of x[i+1], so it is repeated

Reply:
You can either concat up to last node - 1 and add the last one at the end. Or you declare an ndarray and you fill it properly